src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -60,12 +68,107 @@ namespace PowerPreviewSettings
 		}
 	}
 
+	LONG FileExplorerPreviewSettings::SetRegistryValue() const
+	{
+		HKEY hKey = HKEY_CURRENT_USER;
+		const REGSAM WRITE_PERMISSION = KEY_WRITE;
+		DWORD options = 0;
+		HKEY OpenResult;
+
+		LONG err = RegOpenKeyEx(hKey, this->GetSubKey(), options, WRITE_PERMISSION, &OpenResult); 
+
+		if (err == ERROR_SUCCESS)
+		{
+			err = RegSetValueExW(
+					OpenResult,
+					this->GetCLSID(),
+					0,
+					REG_SZ,
+					(LPBYTE)this->GetDisplayName().c_str(),
+					this->GetDisplayName().length() * sizeof(TCHAR));
```

Comment:
why `TCHAR` and not `wchar`? [ref](https://docs.microsoft.com/en-us/office/client-developer/outlook/mapi/tchar)? Are we sure that in all cases `TCHAR` is the same size as `wchar`?

Reply:
Is there are a reason for calculating the size using .length*sizeof(), rather than simply using 
`this->GetDisplayName().size()`, as that would directly return the size in bytes?